opm/parser/eclipse/EclipseState/Schedule/MessagesLimits.hpp

```diff
@@ -0,0 +1,83 @@
+/*
+  Copyright 2016 Statoil ASA.
+
+  This file is part of the Open Porous Media project (OPM).
+
+  OPM is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  OPM is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with OPM.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#ifndef OPM_MESSAGES_HPP
+#define OPM_MESSAGES_HPP
+
+#include <opm/parser/eclipse/EclipseState/Schedule/DynamicState.hpp>
+
+namespace Opm {
+
+    class TimeMap;    
+
+    class MessagesLimits {
+    public:
+
+        /*
+           This constructor will create a new Messages object which is
+           a copy of the input argument, and then all items explicitly
+           set in the record are modified.
+        */
+
+        MessagesLimits(std::shared_ptr<const TimeMap> timemap);
```

Comment:
Technically not if he doesn't use it outside of a reference. A forward declaration is plenty.